traffic_monitor/manager/monitorconfig.go

```diff
@@ -286,25 +307,46 @@ func monitorConfigListen(
 		}
 
 		peerSet := map[tc.TrafficMonitorName]struct{}{}
+		tmsByGroup := make(map[string][]tc.TrafficMonitor)
 		for _, srv := range monitorConfig.TrafficMonitor {
-			if srv.HostName == staticAppData.Hostname {
+			if tc.CacheStatusFromString(srv.ServerStatus) != tc.CacheStatusOnline {
+				continue
+			}
+			tmsByGroup[srv.Location] = append(tmsByGroup[srv.Location], srv)
+		}
+
+		for _, srv := range monitorConfig.TrafficMonitor {
+			if srv.HostName == staticAppData.Hostname || (cfg.DistributedPolling && srv.Location != thisTMGroup) {
 				continue
 			}
 			if tc.CacheStatusFromString(srv.ServerStatus) != tc.CacheStatusOnline {
 				continue
 			}
 			// TODO: the URL should be config driven. -jse
-			url4 := fmt.Sprintf("http://%s:%d/publish/CrStates?raw", srv.IP, srv.Port)
-			url6 := fmt.Sprintf("http://[%s]:%d/publish/CrStates?raw", ipv6CIDRStrToAddr(srv.IP6), srv.Port)
-			peerURLs[srv.HostName] = poller.PollConfig{URL: url4, URLv6: url6, Host: srv.FQDN} // TODO determine timeout.
+			peerURL := fmt.Sprintf("http://%s:%d/publish/CrStates?raw", srv.FQDN, srv.Port)
```

Comment:
Correct, TM peers will no longer oscillate between ipv4 and ipv6, because there is no need. They now just poll each other using their FQDNs (which may resolve to ipv4 or ipv6).